spec/truffle/interop/java_type_name_spec.rb

```diff
@@ -0,0 +1,35 @@
+# Copyright (c) 2018 Oracle and/or its affiliates. All rights reserved. This
+# code is released under a tri EPL/GPL/LGPL license. You can use it,
+# redistribute it and/or modify it under the terms of the:
+#
+# Eclipse Public License version 1.0
+# GNU General Public License version 2
+# GNU Lesser General Public License version 2.1
+
+require_relative '../../ruby/spec_helper'
+
+guard -> { !Truffle.native? } do
+  describe "Truffle::Interop.java_type_name" do
+
+    it "gives a name for a primitive type" do
+      Truffle::Interop.java_type_name(Truffle::Interop.java_type("int")).should == "int"
+    end
+
+    it "gives a name for a primitive type array" do
+      Truffle::Interop.java_type_name(Truffle::Interop.java_type("int[]")).should == "[I"
```

Comment:
Should the name be `int[]` or it's also `[I` on Graal.js?

Reply:
It's compatible with both Graal.js and Nashorn. I'm not aiming to design something new here - just make sure the primitives are there for when people need them for demos. When we have more motivating examples we can suggest changes.